javascript-source/systems/noticeSystem.js

```diff
@@ -73,14 +73,23 @@
             CommandEvent = Packages.tv.phantombot.event.command.CommandEvent,
             notice = $.inidb.get('notices', 'message_' + RandomNotice);
 
+        RandomNotice++;
+
+        if (RandomNotice >= numberOfNotices) {
+            RandomNotice = 0;
+        }
+
         if (notice == null) {
             return;
         }
 
-        RandomNotice++;
+        if (notice.match(/\(gameonly=.*\)/g)) {
+            var game = notice.match(/\(gameonly=(.*)\)/)[1];
 
-        if (RandomNotice >= numberOfNotices) {
-            RandomNotice = 0;
+            if (!$.getGame($.channelName).equalsIgnoreCase(game)) {
+                return sendNotice();
```

Comment:
This seems subject to an infinite loop if all notices are `gameonly` tagged and you are not playing any of the tagged games

Reply:
I thought about that. I assumed people would have at least one hit.  
What do you suggest? Don't call `sendNotice()` again?

Reply:
Maybe use a loop around the function instead of recursive calling
This way you can have a var to track if you've tried them all